static/js/stream_data.js

```diff
@@ -216,8 +229,8 @@ exports.add_sub = function (sub) {
     // We use create_sub_from_server_data at page load.
     // We use create_streams for new streams in live-update events.
 
-    if (!Object.prototype.hasOwnProperty.call(sub, "subscribers")) {
-        sub.subscribers = new LazySet([]);
+    if (!stream_subscribers.has(sub.stream_id)) {
+        exports.set_subscribers(sub, []);
     }
```

Comment:
I guess the theory behind this is that it must be a "new" stream if we don't already have subscriber data for it?  

(I know this block later moves, and this function is only used in tests; I'm msotly wondering if we need more of a comment here).

Reply:
This code is still in a bit of flux, so I will try to improve comments around `peer_data` once everything stabilizes.  For now we at least have this:

~~~
 11 // This maps a stream_id to a LazySet of user_ids who are subscribed.
 12 // We maintain the invariant that this has keys for all all stream_ids
 13 // that we track in the other data structures.  We initialize it during
 14 // clear_subscriptions.